src/Nest/DSL/UpdateSettingsDescriptor.cs

```diff
@@ -1,116 +1,154 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
 using Elasticsearch.Net;
 using Newtonsoft.Json;
 
 namespace Nest
-{
+{ 
+	public class SettingNames
```

Comment:
@Mpdreamz much more better name :+1: . Maybe we should split this class into three? UpdatableSettings, UnmodifiableSettings and UnremovableSettings? https://github.com/elastic/elasticsearch/blob/master/src/main/java/org/elasticsearch/snapshots/RestoreService.java#L92-105